### DISCUSSION TITLE
Shows who purchased the item

### DISCUSSION THREAD

**Discussion by zhu-rengong** [FakeFishGames/Barotrauma#8752] [Ideas and suggestions]
I think it's necessary, when you have wallets, you can adjust crew's salary to make sure each crew member can dispose of their property, then we need to make purchased item show who bought it to avoid the theif stealing.

Alternatively, each cargo box can be marked whose own it is.

**Comment by PokkeTigrex** [comment]
I had the same thoughts, buying stuff and having everyone's order all lump up in one crate might be annoying after spending your marks for it. Perhaps a disposable crate-box instead of the usual metal crates that breaks after the last item is taken out so there isn't tons of metal crates and chem-med crates laying around on the ship on every outpost transition. Another idea is maybe have the metal crate spawn with the person so they immediately have it and they can sort out their purchase?

**Reply by zhu-rengong** [reply]
> I had the same thoughts, buying stuff and having everyone's order all lump up in one crate might be annoying after spending your marks for it. Perhaps a disposable crate-box instead of the usual metal crates that breaks after the last item is taken out so there isn't tons of metal crates and chem-med crates laying around on the ship on every outpost transition. Another idea is maybe have the metal crate spawn with the person so they immediately have it and they can sort out their purchase?

Yes, the more players there are, the more crate will be generated at the beginning of each round, your idea is feasible.
Of course, there are also two options that can be added to the shopping mall: **packing fee** and **expedited fee**
If you decide to purchase many items, packing fee is free, else if you refuse paying, the goods you purchased will be directly transported to the character's inventory instead of the crate.
If you buy few items, expedited fee is free, else if you refuse paying, items you buy will only spawn at the beginning of the next round.
I think this way should solve the problem you mentioned.
I think these two options can try to avoid the situation you mentioned.
